### ISSUE TITLE
Compiler panics with CARGO_INCREMENTAL=1

### ISSUE THREAD

**Issue by MaikKlein** [rust-lang/rust#42237]
https://github.com/MaikKlein/breeze/tree/38baf7a522a503a22ccfbc1fc1b66fe6212c3a56

```
Ubuntu 16.04 

rustc --version --verbose
rustc 1.19.0-nightly (5f3966864 2017-05-25)
binary: rustc
commit-hash: 5f39668642e445fedcf477493f206905dc51c2c8
commit-date: 2017-05-25
host: x86_64-unknown-linux-gnu
release: 1.19.0-nightly
LLVM version: 4.0

```
```
cd examples/gfx
CARGO_INCREMENTAL=1 cargo build --release
```
```
error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/rust-lang/rust/blob/master/CONTRIBUTING.md#bug-reports

note: run with `RUST_BACKTRACE=1` for a backtrace

thread 'rustc' panicked at 'already have hash for FileMap(DefId { krate: CrateNum(20), node: DefIndex(0) => breeze_math/8b623fbab324bdf99d19ede453c6d5ed }, "/home/maik/projects/breeze/examples/gfx/<proc-macro source code>")', /checkout/src/librustc_incremental/persist/hash.rs:254
note: Some details are omitted, run with `RUST_BACKTRACE=full` for a verbose backtrace.
stack backtrace:
   0: std::sys::imp::backtrace::tracing::imp::unwind_backtrace
             at /checkout/src/libstd/sys/unix/backtrace/tracing/gcc_s.rs:49
   1: std::sys_common::backtrace::_print
             at /checkout/src/libstd/sys_common/backtrace.rs:71
   2: std::panicking::default_hook::{{closure}}
             at /checkout/src/libstd/sys_common/backtrace.rs:60
             at /checkout/src/libstd/panicking.rs:355
   3: std::panicking::default_hook
             at /checkout/src/libstd/panicking.rs:365
   4: std::panicking::rust_panic_with_hook
             at /checkout/src/libstd/panicking.rs:549
   5: std::panicking::begin_panic
             at /checkout/src/libstd/panicking.rs:511
   6: std::panicking::begin_panic_fmt
             at /checkout/src/libstd/panicking.rs:495
   7: rustc_incremental::persist::hash::HashContext::load_data
   8: rustc_incremental::persist::hash::HashContext::hash
   9: rustc_incremental::persist::save::save_dep_graph
  10: rustc_driver::driver::phase_4_translate_to_llvm
  11: rustc_driver::driver::compile_input::{{closure}}
  12: rustc_driver::driver::phase_3_run_analysis_passes::{{closure}}
  13: rustc_driver::driver::phase_3_run_analysis_passes
  14: rustc_driver::driver::compile_input
  15: rustc_driver::run_compiler

error: Could not compile `breeze-gfx`.
```

**Comment by MaikKlein** [comment]
Update: 

Going back to `rustup override set nightly-2017-05-15` fixes the problem. The problem must be between 05.15 and 05.26.

**Comment by kennytm** [comment]
Same as #42101? The fix #42175 will be merged soon.

**Comment by michaelwoerister** [comment]
@MaikKlein Thanks for the report! This is indeed the same issue as #42101.
